crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -1187,24 +1721,48 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
 
 pub struct DrawMesh;
 impl<P: PhaseItem> RenderCommand<P> for DrawMesh {
-    type Param = (SRes<RenderAssets<Mesh>>, SRes<RenderMeshInstances>);
-    type ViewQuery = ();
+    type Param = (
+        SRes<RenderAssets<Mesh>>,
+        SRes<RenderMeshInstances>,
+        SRes<PipelineCache>,
+        SRes<PreprocessPipeline>,
+    );
+    type ViewQuery = Has<PreprocessBindGroup>;
     type ItemQuery = ();
     #[inline]
     fn render<'w>(
         item: &P,
-        _view: (),
+        has_preprocess_bind_group: ROQueryItem<Self::ViewQuery>,
         _item_query: Option<()>,
-        (meshes, mesh_instances): SystemParamItem<'w, '_, Self::Param>,
+        (meshes, mesh_instances, pipeline_cache, preprocess_pipeline): SystemParamItem<
+            'w,
+            '_,
+            Self::Param,
+        >,
         pass: &mut TrackedRenderPass<'w>,
     ) -> RenderCommandResult {
+        // If we're using GPU preprocessing, then we're dependent on that
+        // compute shader having been run, which of course can only happen if
+        // it's compiled. Otherwise, our mesh instance data won't be present.
+        if !has_preprocess_bind_group
```

Comment:
increase branch in `drawmesh `could hurt performance when gpu preprocessing disabled , would it be worthwhile to add another RenderCommand like `DrawPreprocessMesh`?

Reply:
That would force us to duplicate all the `DrawFunctions`. I'd rather not do that until we know for sure that performance is a problem. We have yet to see any *actual* performance problems from adding a couple of well predicted branches.